dcicutils/file_utils.py

```diff
@@ -16,43 +25,221 @@ def search_for_file(file: str,
+            else:
+                path = f"{directory}/{file}"
+            files = glob.glob(path, recursive=True if recursive is True else False)
+            if files:
+                for file_found in files:
+                    file_found = os.path.abspath(file_found)
+                    if single is True:
+                        return file_found
+                    if file_found not in files_found:
+                        files_found.append(file_found)
+    if files_found:
+        return files_found[0] if single is True else files_found
+    return None if single is True else []
+
+
+def normalize_path(value: Union[str, pathlib.Path], absolute: bool = False, home: Optional[bool] = None) -> str:
+    """
+    Normalizes the given path value and returns the result; does things like remove redundant
+    consecutive directory separators and redundant parent paths. If the given absolute argument
+    is True than converts the path to an absolute path. If the given home argument is True and
+    if the path can reasonably be represented with a home directory indicator (i.e. "~"), then
+    converts it to such. If the home argument is False and path starts with the home directory
+    indicator then expands it to the actual (absolute) home path. If the given value is not
+    actually even a string (or pathlib.Path) then returns an empty string.
+    """
+    if isinstance(value, pathlib.Path):
+        value = str(value)
+    elif not isinstance(value, str):
+        return ""
+    if not (value := value.strip()) or not (value := os.path.normpath(value)):
+        return ""
+    if home is False:
+        value = os.path.expanduser(value)
+    elif (home is True) and (os.name == "posix") and value.startswith(home := HOME_DIRECTORY + os.sep):
+        value = "~/" + value[len(home):]
+    if absolute is True:
+        value = os.path.abspath(value)
+    return value
+
+
+def get_file_size(file: str) -> Optional[int]:
+    try:
+        return os.path.getsize(file) if isinstance(file, str) else None
+    except Exception:
+        return None
+
+
+def get_file_modified_datetime(file: str) -> Optional[datetime]:
+    try:
+        return datetime.fromtimestamp(os.path.getmtime(file)) if isinstance(file, str) else None
+    except Exception:
+        return None
+
+
+def are_files_equal(filea: str, fileb: str) -> bool:
+    """
+    Returns True iff the contents of the two given files are exactly the same.
+    """
+    try:
+        with open(filea, "rb") as fa:
+            with open(fileb, "rb") as fb:
+                chunk_size = 4096
+                while True:
+                    chunka = fa.read(chunk_size)
+                    chunkb = fb.read(chunk_size)
+                    if chunka != chunkb:
+                        return False
+                    if not chunka:
+                        break
+        return True
+    except Exception:
+        return False
+
+
+def compute_file_md5(file: str) -> str:
```

Comment:
You might want to do a file size check prior to running this as if it's too big it may run for a long time and may want to let users know about it